fix: (platform) schematics for ngx/platform

kavya-b opened the pull request:
#### Please provide a link to the associated issue.
#2812 
#### Please provide a brief summary of this pull request.
Provides ng add schematics for ngx/platform such that running `ng add @fundamental-ngx/platform` will also :
- [x]  install dependent libraries like ngx/core and @angular/localize
- [x]  internally run ng add @fundamental-ngx/core
- [x]  move/merge trans-units from platform lib's translation files into host application's translation files

Provides ng update schematics for ngx/platform such that running `ng update @fundamental-ngx/platform --migrate-only=true --from=0.20.0` will:
- [x]  add localize lib in host application if not already present
- [x]  add/update trans-units from platform lib's translation files into host application's translation files

#### Please try to use this schematic and provide feedback for the same. Following are the steps to run it locally:
**ng-add**
1. Pull this branch feat/platform-schematic 
2. Run the following commands in the fundamental-ngx branch: 
`ng build core` - to build core lib
`ng build platform` - to build platform lib
`npm run build:schematics` - to build core lib schematics
`npm run build:platform-schematics` -  to build platform lib schematics
`npm run sync-versions` - to replace version placeholder tags in core and platform's package.json with proper versions.
3. Create a host application; this could be a new angular project. Ensure that host app is on Ang9 or above as @angular/localize was introduced in Ang9 onwards.
4. In your host application, run the following:
`npm link <path-to-platform-libs-dist-folder>`- this path can be taken from the pulled branch's dist folder->platform->Right click->Copy Path.
`ng add @fundamental-ngx/platform`

**ng-update**

1. Create a host application; this could be a new angular project. 
2. Run `npm i @fundamental-ngx/platform` to have the latest library installed.
3. Follow all steps for `ng-add` so as to have some translations in host application.
4. Make some changes to existing translations in lib, or add new translations by placing i18n markers in lib and running the command `ng xi18n --output-path src/app/platform/locale`
5. Run `npm run build:platform-schematics` in ngx branch. 
6.  In your host application, run the following:
`npm link <path-to-platform-libs-dist-folder>`- this path can be taken from the pulled branch's dist folder->platform->Right click->Copy Path.
7. Run the command `ng update @fundamental-ngx/platform --migrate-only=true --from=0.20.0`
Your updated translations should now reflect in the host app's files. Existing translations, if modified in library, should also have changed in host app.
If you get the error `Repository is not clean. Please commit or stash any changes before updating.`in your host application while doing `ng update`, just commit the files until that point and try `ng update` again.

#### Please check whether the PR fulfills the following requirements
- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

fkolar commented on libs/platform/schematics/ng-add/index.ts:
```diff
@@ -0,0 +1,229 @@
+            callCoreSchematic(_options),
+            addLocalizeLib(_options),
+            readTranslationFiles(_options),
+            endInstallTask()
```
Let's make this install task optional, There are cases where you want to run npm i, yarn i on your own.  Just add additional question to your JSON and schema.

kavya-b replied:
should we make this optional only for core lib, or for all dependent libraries added to package.json by this schematic?

fkolar replied:
This is only for platform.

kavya-b replied:
Sorry, I might have stated it incorrectly. What I meant was platform in turn calls core libs and also calls localize lib, so do we want to make all of these libs optional to be installed?

fkolar replied:
We could right..

kavya-b replied:
yeah, no harm.. will do it

kavya-b replied:
done(for ng add)
also, looks like there is some problem with the version of @angular/schematics. When running core schematics against Ang10 in a different app, it throws the error:
`Cannot find module '@schematics/angular/utility/workspace'`
![image](https://user-images.githubusercontent.com/53509521/88404439-86c09c80-cdeb-11ea-9c2e-9d9e9e8b1dd8.png)
It runs the localize schematic fine, but fails the core schematic.

fkolar replied:
I think there is already opened issue for this

kavya-b replied:
The issue https://github.com/SAP/fundamental-ngx/issues/2835 was closed with the Ang10 migration PR. I have dropped a comment highlighting this issue which occurs only when host application is upgraded from a previous version to Ang10.

kavya-b replied:
@JKMarkowski has responded on https://github.com/SAP/fundamental-ngx/issues/2835 the following: 

> Hi @kavya-b It's because core@0.20 doesn't have angular 10 support. It will be included in 0.21.0

@fkolar Ignoring the above issue for this version as this should get fixed automatically in the next ngx version.

kavya-b commented:
@KevinOkamoto @fkolar added a note about schematic usage in the textarea documentation(file `platform-textarea-docs.component.html`) and moved the languages array in the schematic into its own file(`supported-languages.ts`).
Please review/approve again as I added stuff after Kevin's approval.

fkolar commented:
Can you double check with CI is failing ?

kavya-b commented:
> Can you double check with CI is failing ?

This is from the PR that checks for return types that got pushed through some hours back. Will fix it and rebase once more.
